tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs

```diff
@@ -25,22 +27,23 @@
 using Windows.Foundation;
 using WinUIEx;
 using WinUIEx.Messaging;
+using static System.Runtime.InteropServices.JavaScript.JSType;
 
 namespace DevHome.Environments.ViewModels;
 
 /// <summary>
 /// View model for a compute system. Each 'card' in the UI represents a compute system.
 /// Contains an instance of the compute system object as well.
 /// </summary>
-public partial class ComputeSystemViewModel : ComputeSystemCardBase, IRecipient<ComputeSystemOperationStartedMessage>, IRecipient<ComputeSystemOperationCompletedMessage>
+public partial class ComputeSystemViewModel : ComputeSystemCardBase, IRecipient<ComputeSystemOperationStartedMessage>, IRecipient<ComputeSystemOperationCompletedMessage>, IDisposable
 {
     private readonly ILogger _log = Log.ForContext("SourceContext", nameof(ComputeSystemViewModel));
     private readonly StringResource _stringResource;
     private readonly WindowEx _windowEx;
     private readonly IComputeSystemManager _computeSystemManager;
     private readonly ComputeSystemProvider _provider;
 
-    private readonly object _lock = new();
+    private readonly SemaphoreSlim _semaphoreSlimLock = new(1, 1);

```

Comment:
If it is one thread max, how is it better?

Reply:
It's because you can't use an async method inside a lock. At least that's what the compiler is telling me. Thats the only reason here